packages/dds/tree/src/core/tree/detachedFieldIndex.ts

```diff
@@ -134,9 +155,21 @@ export class DetachedFieldIndex {
 		return key;
 	}
 
-	public deleteEntry(nodeId: Delta.DetachedNodeId): void {
-		const found = deleteFromNestedMap(this.detachedNodeToField, nodeId.major, nodeId.minor);
-		assert(found, 0x7ab /* Unable to delete unknown entry */);
+	public deleteEntry(nodeId: Delta.DetachedNodeId, count: number = 1): void {
+		for (let i = 0; i < count; ++i) {
+			const found = deleteFromNestedMap(
+				this.detachedNodeToField,
+				nodeId.major,
+				nodeId.minor + i,
+			);
+			assert(found, 0x7ab /* Unable to delete unknown entry */);
+		}
+		// Delete the detached node ranges accordingly
+		if (this.detachedNodeRangeMap.has(nodeId.major)) {
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			const rangeMap = this.detachedNodeRangeMap.get(nodeId.major)!;
+			deleteFromRangeMap(rangeMap, nodeId.minor, count);
```

Comment:
The boolean returned by `deleteFromRangeMap` only tells us whether the number of deleted items is 0 or greater than 0. That's not enough for our needs here: we need to know whether all items will be deleted. I think `deleteFromRangeMap` should either return nothing, or return a count that represents the number of deleted items (it's the more useful API and it's also clearer: right now it's very easy to misinterpret what that boolean represents). That said, even returning the number of deleted items is not helpful to us here, because we want to know *before* we delete anthing. This likely means we need a new function that just returns a count of items in a range map, but doesn't change the map in any way.